src/condor_utils/submit_utils.cpp

```diff
@@ -9509,38 +9463,85 @@ process_job_credentials(
 	std::string & URL,
 	std::string & error_string
 ) {
+	std::string token_names;
+	std::vector<ClassAd> token_ads;
+	CredSorter sorter;
+
+	error_string.clear();
+
+	bool add_local = param_boolean("SUBMIT_ADD_LOCAL_CREDMON_PROVIDERS", true);
+
+	if (submit_hash.NeedsOAuthServices(add_local, token_names, &token_ads, &error_string)) {
+		if ( !error_string.empty()) {
+			return 1;
+		}
+	}
+
 	std::string storer;
 	if(param(storer, "SEC_CREDENTIAL_STORER")) {
 		// SEC_CREDENTIAL_STORER is a script to run that calls
 		// condor_store_cred when it has new credentials to store.
 		// Pass it parameters of the service requests needed as
 		// defined in the submit file.
-		ArgList args;
-		args.AppendArg(storer);
-		if( get_oauth_service_requests(args, submit_hash, error_string) ) {
-			if( my_system(args) != 0 ) {
+		// It's used for Vault-managed tokens, so ignore other token
+		// types.
+		bool call_storer = false;
+		ArgList storer_args;
+
+		storer_args.AppendArg(storer);
+
+		for (const auto& request: token_ads) {
+			std::string request_arg;
+			std::string str;
+			request.LookupString("Service", str);
+			if (str.empty()) {
+				continue;
+			}
+			if (sorter.Sort(str) != CredSorter::VaultType) {
```

Comment:
Calling this a sorter when it doesn't actually sort, but instead categorizes/classifies or partitions is weird but whatever; having the names used in the implementation be `cred_name` rather than `service_name` is downright deceptive.

Reply:
In my original design, the caller passed all of the names into the object at once and got back a list of names for each credmon type. Upon implementation, it made more sense to do one name at a time, and I didn't reconsider the name of the class.

We use 'credential', 'token', 'service', 'provider', and 'module' to refer to these things (all of these labels appear in `process_job_credentials()`). In the submit_utils code, I kept changing the names back and forth because the existing code was so inconsistent.
`#include <we_are_bad_at_naming_things>`

I am open to changing some of the names.